core/templates/dev/head/pages/exploration_player/HintButtonDirective.js

```diff
@@ -18,15 +18,39 @@
 
 oppia.directive('hintButton', [
   'UrlInterpolationService', function(UrlInterpolationService) {
+    var SHOW_NEED_HINT_MESSAGE_DELAY = 1000;
+    var NUM_ATTEMPTS_BEFORE_SHOWING_NEED_HINT_MESSAGE = 2;
+
     return {
       restrict: 'E',
       scope: {
         onClickHintButton: '&',
         allHintsAreExhausted: '&',
-        currentHintIsAvailable: '&'
+        currentHintIsAvailable: '&',
       },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
-        'hint_button_directive.html')
+        'hint_button_directive.html'),
+      controller: [
+        '$scope', '$rootScope', '$timeout', 'NumberAttemptsService',
+        function($scope, $rootScope, $timeout, NumberAttemptsService) {
+          $scope.isShowingHintTooltip = false;
+          var showNeedHintIfNecessary = function() {
+            if (NumberAttemptsService.getNumberAttempts() >=
+                NUM_ATTEMPTS_BEFORE_SHOWING_NEED_HINT_MESSAGE) {
+              $timeout(function() {
+                $scope.isShowingHintTooltip = true;
+                $scope.isShowingNeedHintMessage = true;
+              }, SHOW_NEED_HINT_MESSAGE_DELAY);
+            }
+          };
+
+          $scope.hideNeedHintMessage = function() {
+            $scope.isShowingHintTooltip = false;
+            $scope.isShowingNeedHintMessage = false;
+          };
+
+          $scope.$on('oppiaFeedbackAvailable', showNeedHintIfNecessary());
```

Comment:
I think you want to get rid of the () at the end. What you're doing here is executing the showNeedHintIfNecessary() function and passing the return value to $scope.$on().

I think a $broadcast is fine in this context. An alternative is to define

```
    $scope.isHintShown = function() {
      return NumberAttemptsService.getNumberAttempts() >= ...;
    };
```

but that's a little inefficient. It's a fine approach too, though.

Reply:
Ah yes that was the source of my troubles...